trace/trace.go

```diff
@@ -28,29 +28,46 @@ import (
 type traceStep struct {
 	stepTime time.Time
 	msg      string
+	msgArgs  []interface{}
```

Comment:
existing uses that include a % in msg and do not populate args will now output malformed sprintf output, right? should we only sprintf if `len(msgArgs) > 0`?

Reply:
either way, needs godoc about what msg is expected to be (literal string? format string? format string if msgArgs is non-zero-length? etc)

Reply:
Oops. I've fixed by organizing the formatted string more clearly and keeping them separate from the literal strings.

Added a comment as well.